dipy/reconst/dki.py

```diff
@@ -0,0 +1,743 @@
+            # Make sure it's boolean, so that it can be used to mask
+            mask = np.array(mask, dtype=bool, copy=False)
+            data_in_mask = data[mask]
+        else:
+            data_in_mask = data
+
+        params_in_mask = self.fit_method(self.design_matrix, data_in_mask,
+                                         *self.args, **self.kwargs)
+
+        if mask is None:
+            out_shape = data.shape[:-1] + (-1, )
+            dki_params = params_in_mask.reshape(out_shape)
+        else:
+            dki_params = np.zeros(data.shape[:-1] + (27,))
+            dki_params[mask, :] = params_in_mask
+
+        return DiffusionKurtosisFit(self, dki_params)
+
+    def predict(self, dki_params, S0=1):
+        """ Predict a signal for this DKI model class instance given
+        parameters.
+
+        Parameters
+        ----------
+        dki_params : ndarray (x, y, z, 27) or (n, 27)
+            All parameters estimated from the diffusion kurtosis model.
+            Parameters are ordered as follow:
+                1) Three diffusion tensor's eingenvalues
+                2) Three lines of the eigenvector matrix each containing the
+                   first, second and third coordinates of the eigenvector
+                3) Fifteen elements of the kurtosis tensor
+
+        gtab : a GradientTable class instance
+            The gradient table for this prediction
+
+        S0 : float or ndarray (optional)
+            The non diffusion-weighted signal in every voxel, or across all
+            voxels. Default: 1
+        """
+        return dki_prediction(dki_params, self.gtab, S0)
+
+
+class DiffusionKurtosisFit(TensorFit):
+
+    def __init__(self, model, model_params):
+        """ Initialize a DiffusionKurtosisFit class instance.
+
+        Since DKI is an extension of DTI, class instance is defined as a
+        subclass of the TensorFit from dti.py
+        """
+        TensorFit.__init__(self, model, model_params)
+
+    @property
+    def kt(self):
+        """
+        Returns the 15 independent elements of the kurtosis tensor as an array
+        """
+        return self.model_params[..., 12:]
+
+    def akc(self, sphere):
+        r""" Calculate the apparent kurtosis coefficient (AKC) in each
+        direction on the sphere for each voxel in the data
+
+        Parameters
+        ----------
+        sphere : Sphere class instance
+
+        Returns
+        -------
+        akc : ndarray
+           The estimates of the apparent kurtosis coefficient in every
+           direction on the input sphere
+
+        Notes
+        -----
+        The calculation of ADC, relies on the following relationship:
+
+        .. math ::
+
+            ADC = \vec{b} Q \vec{b}^T
+
+        Where Q is the quadratic form of the tensor.
+        """
+        return apparent_kurtosis_coef(self.model_params, sphere)
+
+    def dki_predict(self, gtab, S0=1):
+        r""" Given a DKI model fit, predict the signal on the vertices of a
```

Comment:
More accurate to say that the prediction is made for a gradient table, rather than a sphere